salt/state.py

```diff
@@ -2016,11 +2016,74 @@ def _call_parallel_target(self, name, cdata, low):
             }
 
         utc_finish_time = datetime.datetime.utcnow()
+        timezone_delta = datetime.datetime.utcnow() - datetime.datetime.now()
+        local_finish_time = utc_finish_time - timezone_delta
+        local_start_time = utc_start_time - timezone_delta
+        ret["start_time"] = local_start_time.time().isoformat()
         delta = utc_finish_time - utc_start_time
         # duration in milliseconds.microseconds
         duration = (delta.seconds * 1000000 + delta.microseconds) / 1000.0
         ret["duration"] = duration
 
+        if "retry" in low:
+            retries = 1
+            low["retry"] = self.verify_retry_data(low["retry"])
+            if not sys.modules[self.states[cdata["full"]].__module__].__opts__["test"]:
+                while low["retry"]["attempts"] >= retries:
+
+                    if low["retry"]["until"] == ret["result"]:
+                        break
+
+                    interval = low["retry"]["interval"]
+                    if low["retry"]["splay"] != 0:
+                        interval = interval + random.randint(0, low["retry"]["splay"])
+                    log.info(
+                        "State result does not match retry until value, "
+                        "state will be re-run in %s seconds",
+                        interval,
+                    )
+                    self.functions["test.sleep"](interval)
```

Comment:
Why are we calling the `test.sleep` function here instead of just using python's time's lib with `time.sleep`?

Reply:
No particular reason, most of this code was copied from the existing `retry` functionality.  In this instance it probably makes more sense to use `time.sleep` as this is running inside it's own thread/process.  I'll update it.